app/assets/javascripts/controllers/middleware_topology/middleware_topology_controller.js

```diff
@@ -37,12 +38,12 @@ function MiddlewareTopologyCtrl($scope, $http, $interval, $location, topologySer
     $scope.show_hide_names = function() {
        var vertices = $scope.vs;
 
-       if($scope.checkboxModel.value) {
-            vertices.selectAll("text")
+       if ($scope.checkboxModel.value) {
```

Comment:
How is this change related to adding data source support to topology? if there's another issue that needs fixing here, it is worth to submit it as a separate PR.